### PR TITLE
Bugfix/spk 126. ReadKey flow. Show error when read key with expiry date less than current

### PR DESCRIPTION
**Pull request by AlexMisiulia**

### CODE REVIEW

File: common/src/main/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModel.kt [hunk 1]
```diff
@@ -75,11 +74,19 @@ class ReadNfcViewModel @Inject constructor(
...
-        } else {
+        val keyExpireDateMillis = nfcPersonalFormData.personalForm.keyExpireDate
+        val keyExpireDate = if(keyExpireDateMillis != null) Date(keyExpireDateMillis) else null
+
+        val isKeyExpired = dateUtils.getCurrentDate().roundToDays().after(keyExpireDate?.roundToDays())
+
+        if(!isPinCodeCorrect) {
             setLoading(false)
             _pinCodeIncorrectLiveData.value = Event(true)
+        } else if(isKeyExpired) {
```
Comment (AlexMisiulia) [inline common/src/main/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModel.kt]: Make this statement first

File: common/src/test/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModelTest.kt [hunk 2]
```diff
@@ -36,7 +40,9 @@ class ReadNfcViewModelTest : BaseNfcViewModelTest<ReadNfcViewModel>() {
     @Before
     fun before() {
         loadingObserver = LiveDataTestObserver(viewModel.isLoadingLiveData)
-        whenever(dateUtils.getCurrentDate()).thenReturn(Date())
+
+        val currDate = Date()
```
Comment (AlexMisiulia) [inline common/src/test/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModelTest.kt]: Revert this change. It is useless

File: common/src/main/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModel.kt (hunk 1 above)
Comment (AlexMisiulia) [reply common/src/main/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModel.kt]: Done

File: common/src/test/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModelTest.kt (hunk 2 above)
Comment (AlexMisiulia) [reply common/src/test/java/com/yoti/mobile/android/social/common/features/nfc/readkey/ReadNfcViewModelTest.kt]: Done

File: app/src/test/java/com/yoti/mobile/android/social/keys/features/shared/features/formlist/FormListViewModelTest.kt [hunk 3]
```diff
@@ -16,7 +16,7 @@ private const val SECOND_FORM_NAME = "Form2"
 private const val PROJECT_NAME = "Project"
 private const val SECOND_PROJECT_NAME = "Project2"
 private const val DESCRIPTION = "description"
-private const val KEY_EXPIRE_DATE = "20/05/2019"
+private const val KEY_EXPIRE_DATE = 1558688923544L
```
Comment (cquemin) [inline app/src/test/java/com/yoti/mobile/android/social/keys/features/shared/features/formlist/FormListViewModelTest.kt]: Would be nice to put a comment with date it correspond to

Comment (cquemin) [inline app/src/test/java/com/yoti/mobile/android/social/keys/features/shared/features/formlist/FormListViewModelTest.kt]: Would be nice to put a comment with date it correspond to. (Same in the other classes where it is used)

Comment (AlexMisiulia) [reply app/src/test/java/com/yoti/mobile/android/social/keys/features/shared/features/formlist/FormListViewModelTest.kt]: Done
